src/library/scala/math/Ordering.scala

```diff
@@ -224,11 +224,7 @@ object Ordering extends LowPriorityOrderingImplicits {
   implicit object Unit extends UnitOrdering
 
   trait BooleanOrdering extends Ordering[Boolean] {
-    def compare(x: Boolean, y: Boolean) = (x, y) match {
-      case (false, true) => -1
-      case (true, false) => 1
-      case _ => 0
-    }
+    def compare(x: Boolean, y: Boolean) = if(x) { if (y) 0 else 1} else {if (y) -1 else 0}
```

Comment:
For the record, scalac can now also do it :) will add it as a test case. cfr-decompiler shows
```
    public int compare(boolean x, boolean y) {
        int n = false == x && true == y ? -1 : (true == x && false == y ? 1 : 0);
        return n;
    }
```